scylla/src/transport/topology.rs

```diff
@@ -449,13 +450,52 @@ async fn query_table_schema(
         clustering_key[position as usize] = column_name;
     }
 
+    let partitioner = query_table_partitioner(conn, keyspace_name, table_name)
+        .await
+        .unwrap_or("".to_string());
```

Comment:
I know that the custom partitioner info might not be present (Cassandra won't have this table and, but we shouldn't just ignore all errors. What if the query times out? In such case the user will get incorrect information from the metadata; it's better to fail the whole refresh in such situation instead.

If the `system_schema.scylla_tables` table is not present or there is no `partitioner` column, you will get `DbError::Invalid` - you can detect this and e.g. return `None` in such case.

Reply:
`None`? So do you prefer to store the partitioner as an `Option<String>` rather than `String`?

Reply:
Yes. The lack of a custom partitioner is represented by `null` in the `partitioner` column, so I think it makes sense to use `Option`.